lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -262,4 +298,24 @@ defmodule Mix.Compilers.Elixir do
     # is properly stored.
     Mix.Dep.ElixirSCM.update
   end
+
+  defp entries_from_manifest_map(manifest_map) do
+    for {_, manifest} <- manifest_map,
+        entry <- read_manifest(manifest),
+        do: entry
+  end
+
+  # Since we can have nested paths, we might have duplicates here.
+  defp dedupe_files_map(map) do
```

Comment:
I have a littlr performance concern here. It might be interesting to compile a large project and see what the effect is.

Comparing files is simple and we look for an exact match. I could probably change the logic to look for subdirs by looking for a substring in the path. It's just a little more complicated.